### PR TITLE
Type check for kwargs

### LINKED ISSUE #21: Add type check and conversion for kwargs
Ideally we would add a check to convert user given kwargs to the same type as specified by `::Type{T}`. Something like `gain = gain isa T ? gain : convert(T, gain)`. At the moment we either force the conversion like [here](https://github.com/LuxDL/WeightInitializers.jl/blob/main/src/initializers.jl#L59) or it goes unnoticed like [here](https://github.com/LuxDL/WeightInitializers.jl/blob/main/src/initializers.jl#L78). 

Currently the following code returns a Float64, adding the check would make the workflow less cumbersome since it avoids the user the step of always double checking and converting the kwargs types

```julia
weights_cl = kaiming_normal(Float16; gain=1.0)
weights = weights_cl(2, 5)
#2×5 Matrix{Float64}:
# -0.160808  -0.187664   0.187882  0.918777  -0.136354
#  0.486026   0.321397  -0.30655   0.145306   0.206441
```

I can get this on its way, I just wanted to check if it's something worth implementing.

We also should cover this case in the tests, I don't think the kwargs typing is being tested at the moment

### PR DESCRIPTION
**Pull request by MartinuzziFrancesco**
Addresses #21

### CODE REVIEW

File: ext/WeightInitializersCUDAExt.jl [hunk 1]
```diff
@@ -38,6 +39,7 @@ end
 
 function identity_init(rng::AbstractCuRNG, ::Type{T}, dims::Integer...;
         gain::Number=1, shift::Integer=0) where {T <: Number}
+    gain = gain isa T ? gain : convert(T, gain)
```
Comment (avik-pal) [inline ext/WeightInitializersCUDAExt.jl]: I don't think you need to check for this. The compiler is smart enough to get rid of these

Comment (MartinuzziFrancesco) [reply ext/WeightInitializersCUDAExt.jl]: ~should we also get rid of [this](https://github.com/LuxDL/WeightInitializers.jl/blob/main/src/initializers.jl#L59) then?~

Comment (MartinuzziFrancesco) [reply ext/WeightInitializersCUDAExt.jl]: ~the `T(gain)` specifically~

Comment (MartinuzziFrancesco) [reply ext/WeightInitializersCUDAExt.jl]: sorry scratch that, I misunderstood

Comment (avik-pal) [reply ext/WeightInitializersCUDAExt.jl]: I meant `gain isa T ? gain : convert(T, gain)` can be written as `T(gain)` and the compiler should be smart to remove it if the types match.

Comment (MartinuzziFrancesco) [reply ext/WeightInitializersCUDAExt.jl]: ah I see, should I take that approach with all of the checks?

Comment (avik-pal) [reply ext/WeightInitializersCUDAExt.jl]: yes

**Comment by avik-pal** [comment]
There is some eyntax error, probably `; gain` was written as `; T(gain)` instead of `; gain=T(gain)`

**Comment by MartinuzziFrancesco** [comment]
true, I forgot to fix the `orthogonal` call

**Review by avik-pal — APPROVED** [review]

**Pull request merged**